In [1]:
import numpy as np
import pandas as pd
import sklearn
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn import metrics
from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier,AdaBoostClassifier
from sklearn.model_selection import train_test_split,cross_val_score
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score,auc
from sklearn.preprocessing import LabelEncoder

import warnings
warnings.filterwarnings('ignore')

In [2]:
df=pd.read_excel("covid19_clinical_dataset.xlsx")

In [3]:
pd.set_option('display.max_columns',None)
pd.set_option('display.max_rows',None)

In [4]:
df.head(10)

ID  age    result  regular ward    semi-intensive unit  \
0  44477f75e8169d2   13  negative              0                     0   
1  126e9dd13932f68   17  negative              0                     0   
2  a46b4402a0e5696    8  negative              0                     0   
3  f7d619a94f97c45    5  negative              0                     0   
4  d9e41465789c2b5   15  negative              0                     0   
5  75f16746216c4d1    9  negative              0                     0   
6  2a2245e360808d7   13  negative              0                     0   
7  509197ec73f1400   16  negative              0                     0   
8  8bb9d64f0215244    1  negative              0                     1   
9  5f1ed301375586c   17  negative              0                     0   

    intensive care unit  Hematocrit  Hemoglobin  Platelets  \
0                     0         NaN         NaN        NaN   
1                     0    0.236515   -0.022340  -0.517413   
2                     0         NaN         NaN        NaN   
3                     0         NaN         NaN        NaN   
4                     0         NaN         NaN        NaN   
5                     0         NaN         NaN        NaN   
6                     0         NaN         NaN        NaN   
7                     0         NaN         NaN        NaN   
8                     0   -1.571682   -0.774212   1.429667   
9                     0         NaN         NaN        NaN   

   Mean platelet volume       RBCs  Lymphocytes      MCHC  Leukocytes  \
0                    NaN       NaN          NaN       NaN         NaN   
1               0.010677  0.102004     0.318366 -0.950790    -0.09461   
2                    NaN       NaN          NaN       NaN         NaN   
3                    NaN       NaN          NaN       NaN         NaN   
4                    NaN       NaN          NaN       NaN         NaN   
5                    NaN       NaN          NaN       NaN         NaN   
6                    NaN       NaN          NaN       NaN         NaN   
7                    NaN       NaN          NaN       NaN         NaN   
8              -1.672222 -0.850035    -0.005738  3.331071     0.36455   
9                    NaN       NaN          NaN       NaN         NaN   

   Basophils       MCH  Eosinophils       MCV  Monocytes       RDW  \
0        NaN       NaN          NaN       NaN        NaN       NaN   
1  -0.223767 -0.292269     1.482158  0.166192   0.357547 -0.625073   
2        NaN       NaN          NaN       NaN        NaN       NaN   
3        NaN       NaN          NaN       NaN        NaN       NaN   
4        NaN       NaN          NaN       NaN        NaN       NaN   
5        NaN       NaN          NaN       NaN        NaN       NaN   
6        NaN       NaN          NaN       NaN        NaN       NaN   
7        NaN       NaN          NaN       NaN        NaN       NaN   
8  -0.223767  0.178175     1.018625 -1.336024   0.068652 -0.978899   
9        NaN       NaN          NaN       NaN        NaN       NaN   

   Serum Glucose Respiratory Syncytial Virus   Influenza A   Influenza B  \
0            NaN                         NaN           NaN           NaN   
1      -0.140648                not_detected  not_detected  not_detected   
2            NaN                         NaN           NaN           NaN   
3            NaN                         NaN           NaN           NaN   
4            NaN                not_detected  not_detected  not_detected   
5            NaN                         NaN           NaN           NaN   
6            NaN                         NaN           NaN           NaN   
7            NaN                         NaN           NaN           NaN   
8      -0.413208                not_detected  not_detected  not_detected   
9            NaN                not_detected  not_detected  not_detected   

  Parainfluenza 1 CoronavirusNL63 Rhinovirus/Enterovirus  \
0             NaN             NaN             

In [5]:
df.tail()

ID  age    result  regular ward    semi-intensive unit  \
5639  ae66feb9e4dc3a0    3  positive              0                     0   
5640  517c2834024f3ea   17  negative              0                     0   
5641  5c57d6037fe266d    4  negative              0                     0   
5642  c20c44766f28291   10  negative              0                     0   
5643  2697fdccbfeb7f7   19  positive              0                     0   

       intensive care unit  Hematocrit  Hemoglobin  Platelets  \
5639                     0         NaN         NaN        NaN   
5640                     0         NaN         NaN        NaN   
5641                     0         NaN         NaN        NaN   
5642                     0         NaN         NaN        NaN   
5643                     0    0.694287    0.541564  -0.906829   

      Mean platelet volume       RBCs  Lymphocytes      MCHC  Leukocytes  \
5639                    NaN       NaN          NaN       NaN         NaN   
5640                    NaN       NaN          NaN       NaN         NaN   
5641                    NaN       NaN          NaN       NaN         NaN   
5642                    NaN       NaN          NaN       NaN         NaN   
5643              -0.325903  0.578024    -0.295726 -0.353319   -1.288428   

      Basophils       MCH  Eosinophils       MCV  Monocytes      RDW  \
5639        NaN       NaN          NaN       NaN        NaN      NaN   
5640        NaN       NaN          NaN       NaN        NaN      NaN   
5641        NaN       NaN          NaN       NaN        NaN      NaN   
5642        NaN       NaN          NaN       NaN        NaN      NaN   
5643  -1.140144 -0.135455    -0.835508  0.025985   0.567652 -0.18279   

      Serum Glucose Respiratory Syncytial Virus Influenza A Influenza B  \
5639            NaN                         NaN         NaN         NaN   
5640            NaN                         NaN         NaN         NaN   
5641            NaN                         NaN         NaN         NaN   
5642            NaN                         NaN         NaN         NaN   
5643            NaN                         NaN         NaN         NaN   

     Parainfluenza 1 CoronavirusNL63 Rhinovirus/Enterovirus  \
5639             NaN             NaN                    NaN   
5640             NaN             NaN                    NaN   
5641             NaN             NaN                    NaN   
5642             NaN             NaN                    NaN   
5643             NaN             NaN                    NaN   

      Mycoplasma pneumoniae Coronavirus HKU1 Parainfluenza 3  \
5639                    NaN              NaN             NaN   
5640                    NaN              NaN             NaN   
5641                    NaN              NaN             NaN   
5642                    NaN              NaN             NaN   
5643                    NaN              NaN             NaN   

     Chlamydophila pneumoniae Adenovirus Parainfluenza 4 Coronavirus229E  \
5639                      NaN        NaN             NaN             NaN   
5640                      NaN        NaN             NaN             NaN   
5641                      NaN        NaN             NaN             NaN   
5642                      NaN        NaN             NaN             NaN   
5643                      NaN        NaN             NaN             NaN   

     CoronavirusOC43 Inf A H1N1 2009 Bordetella pertussis Metapneumovirus  \
5639             NaN             NaN                  NaN             NaN   
5640             NaN             NaN                  NaN             NaN   
5641             NaN             NaN                  NaN             NaN   
5642             NaN             NaN                  NaN             NaN   
5643             NaN             NaN                  NaN             NaN   

     Parainfluenza 2  Neutrophils      Urea  Proteina C reativa mg/dL  \
5639             NaN          NaN       NaN                       N

In [6]:
df.isna().sum()

ID                                0
age                               0
result                            0
regular ward                      0
 semi-intensive unit              0
 intensive care unit              0
Hematocrit                     5041
Hemoglobin                     5041
Platelets                      5042
Mean platelet volume           5045
RBCs                           5042
Lymphocytes                    5042
MCHC                           5042
Leukocytes                     5042
Basophils                      5042
MCH                            5042
Eosinophils                    5042
MCV                            5042
Monocytes                      5043
RDW                            5042
Serum Glucose                  5436
Respiratory Syncytial Virus    4290
Influenza A                    4290
Influenza B                    4290
Parainfluenza 1                4292
CoronavirusNL63                4292
Rhinovirus/Enterovirus         4292
Mycoplasma pneumoniae       

In [7]:
df.drop("ID",axis=1,inplace=True)

In [8]:
df.shape

(5644, 110)

In [9]:
df_new=df.interpolate(method ='linear', limit_direction ='forward') 
df_new.head()

age    result  regular ward    semi-intensive unit   intensive care unit  \
0   13  negative              0                     0                     0   
1   17  negative              0                     0                     0   
2    8  negative              0                     0                     0   
3    5  negative              0                     0                     0   
4   15  negative              0                     0                     0   

   Hematocrit  Hemoglobin  Platelets  Mean platelet volume       RBCs  \
0         NaN         NaN        NaN                    NaN       NaN   
1    0.236515   -0.022340  -0.517413               0.010677  0.102004   
2   -0.021799   -0.129751  -0.239259              -0.229737 -0.034001   
3   -0.280112   -0.237161   0.038896              -0.470152 -0.170007   
4   -0.538426   -0.344571   0.317050              -0.710566 -0.306013   

   Lymphocytes      MCHC  Leukocytes  Basophils       MCH  Eosinophils  \
0          NaN       NaN         NaN        NaN       NaN          NaN   
1     0.318366 -0.950790   -0.094610  -0.223767 -0.292269     1.482158   
2     0.272065 -0.339096   -0.029016  -0.223767 -0.225063     1.415939   
3     0.225765  0.272599    0.036578  -0.223767 -0.157857     1.349720   
4     0.179464  0.884293    0.102173  -0.223767 -0.090650     1.283501   

        MCV  Monocytes       RDW  Serum Glucose Respiratory Syncytial Virus  \
0       NaN        NaN       NaN            NaN                         NaN   
1  0.166192   0.357547 -0.625073      -0.140648                not_detected   
2 -0.048410   0.316276 -0.675619      -0.179585                         NaN   
3 -0.263012   0.275005 -0.726166      -0.218522                         NaN   
4 -0.477615   0.233734 -0.776713      -0.257460                not_detected   

    Influenza A   Influenza B Parainfluenza 1 CoronavirusNL63  \
0           NaN           NaN             NaN             NaN   
1  not_detected  not_detected    not_detected    not_detected   
2           NaN           NaN             NaN             NaN   
3           NaN           NaN             NaN             NaN   
4  not_detected  not_detected    not_detected    not_detected   

  Rhinovirus/Enterovirus  Mycoplasma pneumoniae Coronavirus HKU1  \
0                    NaN                    NaN              NaN   
1               detected                    NaN     not_detected   
2                    NaN                    NaN              NaN   
3                    NaN                    NaN              NaN   
4               detected                    NaN     not_detected   

  Parainfluenza 3 Chlamydophila pneumoniae    Adenovirus Parainfluenza 4  \
0             NaN                      NaN           NaN             NaN   
1    not_detected             not_detected  not_detected    not_detected   
2             NaN                      NaN           NaN             NaN   
3             NaN                      NaN           NaN             NaN   
4    not_detected             not_detected  not_detected    not_detected   

  Coronavirus229E CoronavirusOC43 Inf A H1N1 2009 Bordetella pertussis  \
0             NaN             NaN             NaN                  NaN   
1    not_detected    not_detected    not_detected         not_detected   
2             NaN             NaN             NaN                  NaN   
3             NaN             NaN             NaN                  NaN   
4    not_detected    not_detected    not_detected         not_detected   

  Metapneumovirus Parainfluenza 2  Neutrophils      Urea  \
0             NaN             NaN          NaN       NaN   
1    not_detected    not_detected    -0.619086  1.198059   
2             NaN             NaN    -0.548845  1.017292   
3             NaN             NaN    -0.478603  0.836525   
4    not_detected    not_detected    -0.408361  0.655759   

   Proteina C reativa mg/dL  Creatinine  Potassium    Sodium  \
0                       NaN         NaN   

In [10]:
df_new=df_new.fillna(0) 
df_new.head()

age    result  regular ward    semi-intensive unit   intensive care unit  \
0   13  negative              0                     0                     0   
1   17  negative              0                     0                     0   
2    8  negative              0                     0                     0   
3    5  negative              0                     0                     0   
4   15  negative              0                     0                     0   

   Hematocrit  Hemoglobin  Platelets  Mean platelet volume       RBCs  \
0    0.000000    0.000000   0.000000               0.000000  0.000000   
1    0.236515   -0.022340  -0.517413               0.010677  0.102004   
2   -0.021799   -0.129751  -0.239259              -0.229737 -0.034001   
3   -0.280112   -0.237161   0.038896              -0.470152 -0.170007   
4   -0.538426   -0.344571   0.317050              -0.710566 -0.306013   

   Lymphocytes      MCHC  Leukocytes  Basophils       MCH  Eosinophils  \
0     0.000000  0.000000    0.000000   0.000000  0.000000     0.000000   
1     0.318366 -0.950790   -0.094610  -0.223767 -0.292269     1.482158   
2     0.272065 -0.339096   -0.029016  -0.223767 -0.225063     1.415939   
3     0.225765  0.272599    0.036578  -0.223767 -0.157857     1.349720   
4     0.179464  0.884293    0.102173  -0.223767 -0.090650     1.283501   

        MCV  Monocytes       RDW  Serum Glucose Respiratory Syncytial Virus  \
0  0.000000   0.000000  0.000000       0.000000                           0   
1  0.166192   0.357547 -0.625073      -0.140648                not_detected   
2 -0.048410   0.316276 -0.675619      -0.179585                           0   
3 -0.263012   0.275005 -0.726166      -0.218522                           0   
4 -0.477615   0.233734 -0.776713      -0.257460                not_detected   

    Influenza A   Influenza B Parainfluenza 1 CoronavirusNL63  \
0             0             0               0               0   
1  not_detected  not_detected    not_detected    not_detected   
2             0             0               0               0   
3             0             0               0               0   
4  not_detected  not_detected    not_detected    not_detected   

  Rhinovirus/Enterovirus  Mycoplasma pneumoniae Coronavirus HKU1  \
0                      0                    0.0                0   
1               detected                    0.0     not_detected   
2                      0                    0.0                0   
3                      0                    0.0                0   
4               detected                    0.0     not_detected   

  Parainfluenza 3 Chlamydophila pneumoniae    Adenovirus Parainfluenza 4  \
0               0                        0             0               0   
1    not_detected             not_detected  not_detected    not_detected   
2               0                        0             0               0   
3               0                        0             0               0   
4    not_detected             not_detected  not_detected    not_detected   

  Coronavirus229E CoronavirusOC43 Inf A H1N1 2009 Bordetella pertussis  \
0               0               0               0                    0   
1    not_detected    not_detected    not_detected         not_detected   
2               0               0               0                    0   
3               0               0               0                    0   
4    not_detected    not_detected    not_detected         not_detected   

  Metapneumovirus Parainfluenza 2  Neutrophils      Urea  \
0               0               0     0.000000  0.000000   
1    not_detected    not_detected    -0.619086  1.198059   
2               0               0    -0.548845  1.017292   
3               0               0    -0.478603  0.836525   
4    not_detected    not_detected    -0.408361  0.655759   

   Proteina C reativa mg/dL  Creatinine  Potassium    Sodium  \
0                  0.000000    0.000000   

In [11]:
df_new.dtypes

age                              int64
result                          object
regular ward                     int64
 semi-intensive unit             int64
 intensive care unit             int64
Hematocrit                     float64
Hemoglobin                     float64
Platelets                      float64
Mean platelet volume           float64
RBCs                           float64
Lymphocytes                    float64
MCHC                           float64
Leukocytes                     float64
Basophils                      float64
MCH                            float64
Eosinophils                    float64
MCV                            float64
Monocytes                      float64
RDW                            float64
Serum Glucose                  float64
Respiratory Syncytial Virus     object
Influenza A                     object
Influenza B                     object
Parainfluenza 1                 object
CoronavirusNL63                 object
Rhinovirus/Enterovirus   

In [12]:
df_obj=df_new.select_dtypes(include=[np.object])
df_obj.columns

Index(['result', 'Respiratory Syncytial Virus', 'Influenza A', 'Influenza B',
       'Parainfluenza 1', 'CoronavirusNL63', 'Rhinovirus/Enterovirus',
       'Coronavirus HKU1', 'Parainfluenza 3', 'Chlamydophila pneumoniae',
       'Adenovirus', 'Parainfluenza 4', 'Coronavirus229E', 'CoronavirusOC43',
       'Inf A H1N1 2009', 'Bordetella pertussis', 'Metapneumovirus',
       'Parainfluenza 2', 'Influenza B, rapid test', 'Influenza A, rapid test',
       'Strepto A', 'Urine - Esterase', 'Urine - Aspect', 'Urine - pH',
       'Urine - Hemoglobin', 'Urine - Bile pigments', 'Urine - Ketone Bodies',
       'Urine - Nitrite', 'Urine - Urobilinogen', 'Urine - Protein',
       'Urine - Leukocytes', 'Urine - Crystals', 'Urine - Hyaline cylinders',
       'Urine - Granular cylinders', 'Urine - Yeasts', 'Urine - Color'],
      dtype='object')

In [13]:
cato=['result', 'Respiratory Syncytial Virus', 'Influenza A', 'Influenza B',
       'Parainfluenza 1', 'CoronavirusNL63', 'Rhinovirus/Enterovirus',
       'Coronavirus HKU1', 'Parainfluenza 3', 'Chlamydophila pneumoniae',
       'Adenovirus', 'Parainfluenza 4', 'Coronavirus229E', 'CoronavirusOC43',
       'Inf A H1N1 2009', 'Bordetella pertussis', 'Metapneumovirus',
       'Parainfluenza 2', 'Influenza B, rapid test', 'Influenza A, rapid test',
       'Strepto A', 'Urine - Esterase', 'Urine - Aspect', 'Urine - pH',
       'Urine - Hemoglobin', 'Urine - Bile pigments', 'Urine - Ketone Bodies',
       'Urine - Nitrite', 'Urine - Urobilinogen', 'Urine - Protein',
       'Urine - Leukocytes', 'Urine - Crystals', 'Urine - Hyaline cylinders',
       'Urine - Granular cylinders', 'Urine - Yeasts', 'Urine - Color']
for val in cato:
    le=LabelEncoder()
    df_obj[val]=le.fit_transform(df_obj[val].astype("str"))
df_obj.head()

result  Respiratory Syncytial Virus  Influenza A  Influenza B  \
0       0                            0            0            0   
1       0                            2            2            2   
2       0                            0            0            0   
3       0                            0            0            0   
4       0                            2            2            2   

   Parainfluenza 1  CoronavirusNL63  Rhinovirus/Enterovirus  Coronavirus HKU1  \
0                0                0                       0                 0   
1                2                2                       1                 2   
2                0                0                       0                 0   
3                0                0                       0                 0   
4                2                2                       1                 2   

   Parainfluenza 3  Chlamydophila pneumoniae  Adenovirus  Parainfluenza 4  \
0                0                         0           0                0   
1                2                         2           2                2   
2                0                         0           0                0   
3                0                         0           0                0   
4                2                         2           2                2   

   Coronavirus229E  CoronavirusOC43  Inf A H1N1 2009  Bordetella pertussis  \
0                0                0                0                     0   
1                2                2                2                     2   
2                0                0                0                     0   
3                0                0                0                     0   
4                2                2                2                     2   

   Metapneumovirus  Parainfluenza 2  Influenza B, rapid test  \
0                0                0                        0   
1                2                1                        1   
2                0                0                        0   
3                0                0                        0   
4                2                1                        0   

   Influenza A, rapid test  Strepto A  Urine - Esterase  Urine - Aspect  \
0                        0          0                 0               0   
1                        1          0                 0               0   
2                        0          0                 0               0   
3                        0          0                 0               0   
4                        0          0                 0               0   

   Urine - pH  Urine - Hemoglobin  Urine - Bile pigments  \
0           0                   0                      0   
1           0                   0                      0   
2           0                   0                      0   
3           0                   0                      0   
4           0                   0                      0   

   Urine - Ketone Bodies  Urine - Nitrite  Urine - Urobilinogen  \
0                      0                0                     0   
1                      0                0                     0   
2                      0                0                     0   
3                      0                0                     0   
4                      0                0                     0   

   Urine - Protein  Urine - Leukocytes  Urine - Crystals  \
0                0                   0                 0   
1                0                   0                 0   
2                0                   0                 0   
3                0                   0                 0   
4                0                   0                 0   

   Urine - Hyaline cylinders  Urine - Granular cylinders  Urine - Yeasts  \
0                          0                           0               0   
1                          0                           0      

In [14]:
df_int=df_new.select_dtypes(include=[np.number])

In [15]:
covid19=pd.concat([df_int,df_obj],axis=1)

In [16]:
covid19.head()

age  regular ward    semi-intensive unit   intensive care unit  Hematocrit  \
0   13              0                     0                     0    0.000000   
1   17              0                     0                     0    0.236515   
2    8              0                     0                     0   -0.021799   
3    5              0                     0                     0   -0.280112   
4   15              0                     0                     0   -0.538426   

   Hemoglobin  Platelets  Mean platelet volume       RBCs  Lymphocytes  \
0    0.000000   0.000000               0.000000  0.000000     0.000000   
1   -0.022340  -0.517413               0.010677  0.102004     0.318366   
2   -0.129751  -0.239259              -0.229737 -0.034001     0.272065   
3   -0.237161   0.038896              -0.470152 -0.170007     0.225765   
4   -0.344571   0.317050              -0.710566 -0.306013     0.179464   

       MCHC  Leukocytes  Basophils       MCH  Eosinophils       MCV  \
0  0.000000    0.000000   0.000000  0.000000     0.000000  0.000000   
1 -0.950790   -0.094610  -0.223767 -0.292269     1.482158  0.166192   
2 -0.339096   -0.029016  -0.223767 -0.225063     1.415939 -0.048410   
3  0.272599    0.036578  -0.223767 -0.157857     1.349720 -0.263012   
4  0.884293    0.102173  -0.223767 -0.090650     1.283501 -0.477615   

   Monocytes       RDW  Serum Glucose  Mycoplasma pneumoniae  Neutrophils  \
0   0.000000  0.000000       0.000000                    0.0     0.000000   
1   0.357547 -0.625073      -0.140648                    0.0    -0.619086   
2   0.316276 -0.675619      -0.179585                    0.0    -0.548845   
3   0.275005 -0.726166      -0.218522                    0.0    -0.478603   
4   0.233734 -0.776713      -0.257460                    0.0    -0.408361   

       Urea  Proteina C reativa mg/dL  Creatinine  Potassium    Sodium  \
0  0.000000                  0.000000    0.000000   0.000000  0.000000   
1  1.198059                 -0.147895    2.089928  -0.305787  0.862512   
2  1.017292                 -0.167765    1.528707  -0.129243  0.811172   
3  0.836525                 -0.187635    0.967485   0.047301  0.759832   
4  0.655759                 -0.207505    0.406263   0.223845  0.708492   

   Alanine transaminase  Aspartate transaminase  Gamma-glutamyltransferase   \
0                   0.0                     0.0                         0.0   
1                   0.0                     0.0                         0.0   
2                   0.0                     0.0                         0.0   
3                   0.0                     0.0                         0.0   
4                   0.0                     0.0                         0.0   

   Total Bilirubin  Direct Bilirubin  Indirect Bilirubin  \
0              0.0               0.0                 0.0   
1              0.0               0.0                 0.0   
2              0.0               0.0                 0.0   
3              0.0               0.0                 0.0   
4              0.0               0.0                 0.0   

   Alkaline phosphatase  Ionized calcium   Magnesium  pCO2   Hb saturation  \
0                   0.0               0.0        0.0    0.0            0.0   
1                   0.0               0.0        0.0    0.0            0.0   
2                   0.0               0.0        0.0    0.0            0.0   
3                   0.0               0.0        0.0    0.0            0.0   
4                   0.0               0.0        0.0    0.0            0.0   

   Base excess  pO2  Fio2   Total CO2    pH  HCO3  Rods  Segmented  \
0          0.0  0.0    0.0         0.0  0.0   0.0   0.0        0.0   
1          0.0  0.0    0.0         0.0  0.0   0.0   0.0        0.0   
2          0.0  0.0    0.0         0.0  0.0   0.0   0.0        0.0   
3          0.0  0.0    0.0         0.0  0.0   0.0   0.0        0.0   
4          0.0  0.0    0.0         0.0  0.0   0.0   0.0        0.0   

   Promyel

In [17]:
covid19.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5644 entries, 0 to 5643
Columns: 110 entries, age to Urine - Color
dtypes: float64(70), int32(36), int64(4)
memory usage: 4.0 MB


In [18]:
covid19.skew()

age                             0.034623
regular ward                    8.276088
 semi-intensive unit           10.485579
 intensive care unit           11.607647
Hematocrit                     -0.884681
Hemoglobin                     -0.720533
Platelets                       2.034837
Mean platelet volume            0.401275
RBCs                           -0.480655
Lymphocytes                     0.244508
MCHC                           -0.224457
Leukocytes                      1.153063
Basophils                       2.426110
MCH                            -0.780894
Eosinophils                     2.499692
MCV                            -0.561471
Monocytes                       0.816492
RDW                             2.365056
Serum Glucose                   3.592828
Mycoplasma pneumoniae           0.000000
Neutrophils                     0.100010
Urea                            3.271803
Proteina C reativa mg/dL        3.096117
Creatinine                      0.632105
Potassium       

In [19]:
x= covid19.drop(['result'], axis = 1)

y =covid19['result']

In [20]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X= sc.fit_transform(x)

In [21]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state= 52)

In [22]:
lr= LogisticRegression()
tree = DecisionTreeClassifier()
knn =  KNeighborsClassifier()
svc = SVC()
forest = RandomForestClassifier()
ada = AdaBoostClassifier()
gnb = GaussianNB()

In [23]:
models = {lr:'LogisticRegression',
          tree:'DecisionTree',
          knn: 'KNN',
          svc: 'SVC',
          forest: 'RandomForest',
          ada: 'Adaboost',
          gnb: 'Naive'}

In [24]:
def train(algo,name,X_train,y_train,X_test,y_test):
    algo.fit(X_train,y_train)
    y_pred = algo.predict(X_test)
    score = accuracy_score(y_test,y_pred)
    print(f"--------------------------------------------{name}---------------------------------------------------")
    print(f"Accuracy Score for {name}: {score*100:.4f}%")
    return y_test,y_pred,score

def acc_res(y_test,y_pred):
    print("Confusion Matrix")
    matrix = confusion_matrix(y_test,y_pred)
    print(matrix)

    print("Classification Report")
    print(classification_report(y_test,y_pred))
    
def main(models):
    accuracy_scores = []
    for algo,name in models.items():
        y_test_train,y_pred,acc_score = train(algo,name,X_train,y_train,X_test,y_test)
        acc_res(y_test_train,y_pred)
        accuracy_scores.append(acc_score)
        
    return accuracy_scores
    
accuracy_scores = main(models)

--------------------------------------------LogisticRegression---------------------------------------------------
Accuracy Score for LogisticRegression: 89.9087%
Confusion Matrix
[[1652   17]
 [ 171   23]]
Classification Report
              precision    recall  f1-score   support

           0       0.91      0.99      0.95      1669
           1       0.57      0.12      0.20       194

    accuracy                           0.90      1863
   macro avg       0.74      0.55      0.57      1863
weighted avg       0.87      0.90      0.87      1863

--------------------------------------------DecisionTree---------------------------------------------------
Accuracy Score for DecisionTree: 85.3999%
Confusion Matrix
[[1550  119]
 [ 153   41]]
Classification Report
              precision    recall  f1-score   support

           0       0.91      0.93      0.92      1669
           1       0.26      0.21      0.23       194

    accuracy                           0.85      1863
   macro av

In [25]:
from sklearn.model_selection import cross_val_score
cross_validation = cross_val_score(estimator =ada, X = X_train,y = y_train, cv = 10)
print("Cross validation accuracy of AdaBoostClassifier  = ", cross_validation)
print("\nCross validation mean accuracy of AdaBoostClassifier = ", cross_validation.mean())

Cross validation accuracy of AdaBoostClassifier  =  [0.8944591  0.88390501 0.91029024 0.8944591  0.9021164  0.88888889
 0.9021164  0.89124668 0.91246684 0.8938992 ]

Cross validation mean accuracy of AdaBoostClassifier =  0.8973847881681565


I will choose AdaBoostClassifier as my final model because it perform well in this dataset.

Accuracy_score=90.1235%

Cross_val_score= 0.8973847881681565

In [27]:
from sklearn.externals import joblib

joblib.dump(ada,'covid19_joblib')

['covid19_joblib']